sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -1505,6 +1511,10 @@ private RxDocumentServiceRequest addBatchHeaders(RxDocumentServiceRequest reques
             request.getHeaders().put(HttpConstants.HttpHeaders.AUTHORIZATION, authorization);
         }
 
+        if (this.apiType != null)   {
+            request.getHeaders().put(HttpConstants.HttpHeaders.API_TYPE, this.apiType.toString());
```

Comment:
in addition to the GW mode, if you set it here it will set the header for the direct mode when sending to the BE replicas.

is this intended?

Reply:
Yes this is.